core/src/main/java/org/mule/transformer/simple/AbstractAddVariablePropertyTransformer.java

```diff
@@ -90,9 +90,9 @@ public Object clone() throws CloneNotSupportedException
 
     public void setIdentifier(String identifier)
     {
-        if (identifier == null)
+        if (identifier == null || StringUtils.isBlank(identifier))
```

Comment:
this is redundant. Stringtils.isBlack() already accounts for the value being null